substrate/client/transaction-pool/src/fork_aware_txpool/fork_aware_txpool.rs

```diff
@@ -1395,6 +1395,84 @@ where
 		}
 	}
 
+	/// Fetches the 'provides' tags associated to transactions part of blocks
+	/// of an enacted fork.
+	///
+	/// The 'provides' tags of transactions from enacted blocks are searched
+	/// in inactive views. These tags serve as cache based on the inactive views,
+	/// since prunning transactions is based on their 'provides' tags, which if
+	/// not found in the enacted fork tip active view, then they will be obtained
+	/// by revalidating the transaction, which takes time.
+	async fn provides_tags_from_inactive_views(
+		&self,
+		tree_route: &TreeRoute<Block>,
+	) -> HashMap<ExtrinsicHash<ChainApi>, Vec<Tag>> {
+		// For every enacted block, take its txs.
+		let blocks_xts = future::join_all(
+			tree_route
+				.enacted()
+				.iter()
+				.map(|hn| {
+				    let api = self.api.clone();
+					async move {
+						api
+							.block_body(hn.hash)
+							.await
+							.unwrap_or_else(|e| {
+								log::warn!(target: LOG_TARGET, "provides_tags_from_inactive_views: block_body error request: {}", e);
+								None
+							})
+							.unwrap_or_default()
+					}
+				}))
+			.await
+			.into_iter()
+			.collect::<Vec<_>>();
+
+		// For each enacted block transaction, look for its provides tags in
+		// inactive views of blocks on the retracted fork. It is possible that
+		// transactions of the enacted block to show up in multiple inactive
+		// views, but we'll consider the tags from most recent inactive view.
+		let mut provides_tags_map = HashMap::new();
```

Comment:
how about new function in view store?
Code would become a bit more readable?
```suggestion
        let views = tree_route.retracted.iter().skip(1).map(|h| h.hash).collecet::<Vec<_>>();
		let provides_tags_map = self.view_store.get_provides_tags_from_views(blocks_xts, views);
```

Reply:
Please note that view store could accept transaction hashes.